modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java

```diff
@@ -51,52 +46,65 @@
 public class OperationMgtService {
 
     private static Log log = LogFactory.getLog(OperationMgtService.class);
+    private static final String ACCEPT = "Accept";
 
     @PUT
     @Produces(MediaType.APPLICATION_JSON)
     @Path("{id}")
-    public List<Operation> getPendingOperations
-            (@HeaderParam("Accept") String acceptHeader, @PathParam("id") String id,
-                    List<? extends Operation> resultOperations) {
+    public List<? extends Operation> getPendingOperations
+            (@HeaderParam(ACCEPT) String acceptHeader, @PathParam("id") String id,
+             List<? extends Operation> resultOperations) {
+        Message message;
+        MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
 
-        if (log.isDebugEnabled()) {
-            log.debug("Invoking Android pending operations:" + id);
+        if (id == null || id.isEmpty()) {
+            String errorMessage = "Device identifier is null or empty, hence returning device not found";
+            message = Message.responseMessage(errorMessage).
+                    responseCode(Response.Status.BAD_REQUEST.toString()).build();
+            log.error(errorMessage);
+            throw new AndroidOperationException(message, responseMediaType);
         }
-        Message message = new Message();
-        MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
 
+        DeviceIdentifier deviceIdentifier = AndroidAPIUtils.convertToDeviceIdentifierObject(id);
         try {
-            if (resultOperations != null) {
+            if (!AndroidDeviceUtils.isValidDeviceIdentifier(deviceIdentifier)) {
+                String errorMessage = "Device not found for identifier '" + id + "'";
+                message = Message.responseMessage(errorMessage).
+                        responseCode(Response.Status.BAD_REQUEST.toString()).build();
+                log.error(errorMessage);
+                throw new AndroidOperationException(message, responseMediaType);
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("Invoking Android pending operations:" + id);
+            }
+            if (resultOperations != null && !resultOperations.isEmpty()) {
                 updateOperations(id, resultOperations);
             }
         } catch (OperationManagementException e) {
-            message.setResponseMessage("Issue in retrieving operation management service instance");
-            log.error(message.getResponseMessage(), e);
+            log.error("Issue in retrieving operation management service instance", e);
+        } catch (PolicyComplianceException e) {
+            log.error("Issue in updating Monitoring operation");
+        } catch (DeviceManagementException e) {
+            log.error("Issue in retrieving device management service instance", e);
```

Comment:
No its not, Because this part only handle the update the operation responses and after that executes, device has to fetch its pending operations as well. If we throw an exception here, device will not fetch its pending operations.